clouddrift/analysis.py

```diff
@@ -150,8 +151,12 @@ def segment(
     array([2, 2, 2, 2])
     """
 
-    if type(tolerance) == timedelta:
-        positive_tol = tolerance >= timedelta(seconds=0)
+    # for compatibility with datetime list or np.timedelta64 arrays
+    if type(tolerance) in [np.timedelta64, timedelta]:
```

Comment:
Should check that if the tolerance is a type of timedelta, but the array is not, it should return `AttributeError`.

Reply:
I think it's better to not check that and let the incompatible comparison to raise an exception.

Suppose that a library introduces a new `datetime` class, perhaps `xr.datetime` (it wouldn't be the craziest thing to happen given how many datetimes and timedeltas there already is). If `xr.datetime` is `np.diff` compatible and also comparison-compatible with `pd.Timedelta`, then it will just work. And that was the original intent of Python's duck typing.

Reply:
Sounds good.